test/integration/022_bigquery_test/test_bigquery_execution_project.py

```diff
@@ -0,0 +1,21 @@
+import os
+from test.integration.base import DBTIntegrationTest, use_profile
+
+
+class TestCaseSensitiveModelBigQueryRun(DBTIntegrationTest):
+    @property
+    def schema(self):
+        return "bigquery_test_022"
+
+    @property
+    def models(self):
+        return "execution-project-models"
+
+    @use_profile('bigquery_alternate')
+    def test__bigquery_execute_project(self):
+        results = self.run_dbt(['run', '--models', 'model'])
+        self.assertEqual(len(results), 1)
+        execution_project = os.environ['BIGQUERY_TEST_ALT_DATABASE']
+        self.run_dbt(['test', '--vars',
+                      '{ project_id: %s, job_id: %s }'%(execution_project, "")],
+                     expect_pass=True)
```

Comment:
It isn't possible to access the `job_id`. If anonymous usage tracking were turned on, you could get the `invocation_id` (which is persisted as a job label), but we have it turned off for integration tests.

So I think your best bet is to pass in `self.unique_schema()` and check for something like:
```sql
select project_id from `region-us`.INFORMATION_SCHEMA.JOBS_BY_PROJECT
where date(creation_time) = current_date
  and destination_table.dataset_id = 'test16308377135774383339_bigquery_test_022'
```
And make sure that project_id matches `self.alternative_database`. Whether you'd rather perform that comparison in a custom generic (dbt) test, or use the arbitrary SQL runner (`run_sql_bigquery`) and perform an `assert` check in the python code here, is up to you.

Reply:
OK I tried to use `self.unique_schema()`, I'm not confident it's meant to be used, is it what you had in mind?
I wented for a custom generic DBT test as I felt it make it "reusable" (though I hardly think at this point we would have to with the same macro).